pkg/receive/multitsdb.go

```diff
@@ -302,3 +304,106 @@ func (t *MultiTSDB) TenantAppendable(tenantID string) (Appendable, error) {
 	}
 	return tenant.readyStorage(), nil
 }
+
+// ErrNotReady is returned if the underlying storage is not ready yet.
+var ErrNotReady = errors.New("TSDB not ready")
+
+// ReadyStorage implements the Storage interface while allowing to set the actual
+// storage at a later point in time.
+type ReadyStorage struct {
```

Comment:
It's because it is private now? Can you add `TODO` to upstream making it back public on Prometheus or something?

Reply:
It is removed upstream :( so had to copy it from the old version

Reply:
Noooo it's not removed. It's part of main.go which is something we can pull back to reuse in Prometheus (|:

Reply:
Oh, I did not know that, I will add a TODO